src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxePsiFieldImpl.java

```diff
@@ -213,4 +208,16 @@ public boolean hasModifierProperty(@HaxePsiModifier.ModifierConstant @NonNls @No
     return this.getModifierList().hasModifierProperty(name);
   }
 
+  @NotNull
+  @Override
+  public SearchScope getUseScope() {
+    final PsiElement localVar = UsefulPsiTreeUtil.getParentOfType(this, HaxeLocalVarDeclaration.class);
+    if(localVar != null) {
+      final PsiElement outerBlock = UsefulPsiTreeUtil.getParentOfType(localVar, HaxeBlockStatement.class);
```

Comment:
It doesn't work because `a` inside `for` is not local-variable-declaration. For `for` statement I've added support for renaming from using to declaration. It was missing NameIdentifier. Look new commit, your example will work correctly.